specs/derivation.md

```diff
@@ -679,74 +694,109 @@ place. This section explains how this happens.
 
 Let:
 
-- `refL2` be the (hash of) the current [safe L2 head][g-unsafe-l2-head]
-- `finalizedRef` be the (hash of) the [finalized L2 head][g-finalized-l2-head]: the highest L2 block that can be fully
-  derived from *[finalized][finality]* L1 blocks — i.e. L1 blocks older than two L1 epochs (64 L1 [time
-  slots][g-time-slot]).
+- `safeL2Head` be a variable in the state of the execution engine, tracking the (hash of) the current [safe L2
+  head][g-safe-l2-head]
+- `unsafeL2Head` be a variable in the state of the execution engine, tracking the (hash of) the current [unsafe L2
+  head][g-unsafe-l2-head]
+- `finalizedL2Head` be a variable in the state of the execution engine, tracking the (hash of) the current [finalized L2
+  head][g-finalized-l2-head]
+  - This is not yet implemented, and currently always holds the zero hash — this does not prevent the pseudocode below
+      from working.
 - `payloadAttributes` be some previously derived [payload attributes][g-payload-attr] for the L2 block with number
-  `l2Number(refL2) + 1`
+  `l2Number(safeL2Head) + 1`
 
 [finality]: https://hackmd.io/@prysmaticlabs/finality
 
 Then we can apply the following pseudocode logic to update the state of both the rollup driver and execution engine:
 
 ```javascript
-// request a new execution payload
-forkChoiceState = {
-    headBlockHash: refL2,
-    safeBlockHash: refL2,
-    finalizedBlockHash: finalizedRef,
-}
-[status, payloadID, rpcErr] = engine_forkchoiceUpdatedV1(forkChoiceState, payloadAttributes)
-if (rpcErr != null) soft_error()
-if (status != "VALID") payload_error()
 
-// retrieve and execute the execution payload
-[executionPayload, rpcErr] = engine_getPayloadV1(payloadID)
-if (rpcErr != null) soft_error()
 
-[status, rpcErr] = engine_newPayloadV1(executionPayload)
-if (rpcErr != null) soft_error()
-if (status != "VALID") payload_error()
+fun makeL2Block(payloadAttributes) {
```

Comment:
I prefer using python like syntax throughout the specs but as long as its consistent

Reply:
I'm pretty agnostic about it. Feel free to make a PR to change it, DM me and I'll approve it!